lib/rules/no-unneeded-ternary.js

```diff
@@ -134,7 +134,20 @@ module.exports = {
                         node,
                         loc: node.consequent.loc.start,
                         message: "Unnecessary use of conditional expression for default assignment.",
-                        fix: fixer => fixer.replaceText(node, `${astUtils.getParenthesisedText(sourceCode, node.test)} || ${astUtils.getParenthesisedText(sourceCode, node.alternate)}`)
+                        fix: fixer => {
+                            let nodeTest = astUtils.getParenthesisedText(sourceCode, node.test);
+                            let nodeAlternate = astUtils.getParenthesisedText(sourceCode, node.alternate);
+
+                            if (node.test.type === "ConditionalExpression") {
+                                nodeTest = `(${nodeTest})`;
+                            }
```

Comment:
Ah yes, I thought the same and played around, but it seems it will never happen.